src/rpcblockchain.cpp

```diff
@@ -877,10 +883,15 @@ bool AdvertiseBeacon(std::string &sOutPrivKey, std::string &sOutPubKey, std::str
             if(!BackupConfigFile(GetBackupFilename("gridcoinresearch.conf")))
             {
                 sError = "Failed to backup old configuration file. Beacon not sent.";
-                return true;
+                return false;
             }
             // Backup config with new keys with beacon suffix
-            StoreBeaconKeys(GlobalCPUMiningCPID.cpid, sOutPubKey, sOutPrivKey);
+            if (!StoreBeaconKeys(GlobalCPUMiningCPID.cpid, sOutPubKey, sOutPrivKey))
```

Comment:
Here we store the keys before doing a backup :)

Reply:
just above it backs up the config files and contains old keys. then stores keys and then makes a new backup containing new keys? am i missing something here

Reply:
Ahh, you're right. I think we can remove the second backup. It might even pose an issue since the target filename is with a seconds precision, so the second file is likely to have the same filename as the first.

Reply:
PSST second one has a beacon suffix ;)

Reply:
doh, leave it as is :D